src/lib/store.js

```diff
@@ -1,98 +1,54 @@
+import {
+	writeConsentCookie,
+	decodeConsentData,
+	encodeConsentData
+} from "./cookie/cookie";
 import config from './config';
 import { findLocale } from './localize';
 import log from './log';
-const arrayFrom = require('core-js/library/fn/array/from');
+import { GVL, TCModel, Vector } from '@iabtcf/core';
 
 export const SECTION_INTRO = 0;
 export const SECTION_DETAILS = 1;
 export const SECTION_PURPOSES = 0;
 export const SECTION_VENDORS = 1;
 
-/**
- * Copy a data object and make sure to replace references
- * of Set objects with new ones.
- */
-function copyData(dataObject) {
-	if (typeof dataObject !== 'object') {
-		return dataObject;
-	}
-	const copy = {...dataObject};
-	for (let key in copy) {
-		if (copy.hasOwnProperty(key) && copy[key] instanceof Set) {
-			copy[key] = new Set(copy[key]);
-		}
-	}
-	return copy;
-}
-
 export default class Store {
 	constructor({
-		cmpId = 1,
-		cmpVersion = 1,
-		cookieVersion = 1,
-		vendorConsentData,
-		globalVendorConsentData,
-		publisherConsentData,
-		vendorList,
-		customPurposeList,
-		pubVendorsList,
-		allowedVendorIds
+		cmpId = 280,
+		cmpVersion = 2,
+		cookieVersion = 2,
+		consentData
 	} = {}) {
 		// Keep track of data that has already been persisted
-		this.persistedVendorConsentData = copyData(vendorConsentData);
-		this.persistedPublisherConsentData = copyData(publisherConsentData);
-		this.persistedGlobalVendorConsentData = copyData(globalVendorConsentData);
-
+		consentData = consentData || {};
+		this.persistedConsentData = consentData;
 		const consentLanguage = findLocale().substr(0, 2).toUpperCase();
-		this.vendorConsentData = Object.assign(
-			{
-				selectedPurposeIds: new Set(),
-				selectedVendorIds: new Set()
-			},
-			vendorConsentData,
+		const tcModel = new TCModel();
+		tcModel.cmpId = cmpId;
+		tcModel.cmpVersion = cmpVersion;
+		tcModel.isServiceSpecific = true;
+		tcModel.supportOOB = false;
+
+		this.tcModel = Object.assign(
+			tcModel,
+			consentData,
 			{
 				cookieVersion,
 				cmpId,
 				cmpVersion,
 				consentLanguage
 			});
 
-		this.globalVendorConsentData = Object.assign(
-			{
-				selectedPurposeIds: new Set(),
-				selectedVendorIds: new Set()
-			},
-			globalVendorConsentData,
-			{
-				cookieVersion,
-				cmpId,
-				cmpVersion,
-				consentLanguage
-			}
-		);
-
-		this.publisherConsentData = Object.assign(
-			{
-				selectedStandardPurposeIds: new Set(),
-				selectedCustomPurposeIds: new Set()
-			},
-			publisherConsentData,
-			{
-				cookieVersion,
-				cmpId
-			});
-		this.globalVendorIdsPresentOnList = new Map();
-		this.pubVendorsList = pubVendorsList;
-		this.allowedVendorIds = new Set(allowedVendorIds);
 		this.isConsentToolShowing = false;
 		this.isFooterShowing = false;
 		this.section = SECTION_INTRO;
 		this.subsection = SECTION_PURPOSES;
 		this.hasInitialVendorsRejectionOccured = false;
```

Comment:
As far as I can see this property and `initialVendorsRejection` method are no longer used.

Reply:
initialVendorsRejecton is used within vendors component.

Reply:
You are right. I've checked only *.js files